File: OsmAndWear/src/net/osmand/plus/card/color/palette/main/IColorsPaletteController.java
package net.osmand.plus.card.color.palette.main;

import android.view.View;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;

import net.osmand.plus.base.dialog.interfaces.controller.IDialogController;
import net.osmand.plus.card.color.palette.main.data.PaletteColor;
import net.osmand.plus.card.color.palette.main.data.PaletteSortingMode;
import net.osmand.plus.track.fragments.controller.ColorPickerDialogController.ColorPickerListener;

import java.util.List;

public interface IColorsPaletteController extends IDialogController, ColorPickerListener {

	String ALL_COLORS_PROCESS_ID = "show_all_colors_palette";

	void bindPalette(@NonNull IColorsPalette palette);

	void unbindPalette(@NonNull IColorsPalette palette);

	default void onAllColorsScreenClosed() { }

	void setPaletteListener(@NonNull OnColorsPaletteListener onColorsPaletteListener);

	@ColorInt
	int getControlsAccentColor(boolean nightMode);

	boolean isAccentColorCanBeChanged();

	void onSelectColorFromPalette(@NonNull PaletteColor color, boolean renewLastUsedTime);

	void selectColor(@Nullable Integer color);

	void selectColor(@Nullable PaletteColor paletteColor);

	void onColorLongClick(@NonNull FragmentActivity activity, @NonNull View view, @NonNull PaletteColor color, boolean nightMode);

	void onAllColorsButtonClicked(@NonNull FragmentActivity activity);

	void onAddColorButtonClicked(@NonNull FragmentActivity activity);

	@NonNull
	List<PaletteColor> getColors(@NonNull PaletteSortingMode sortingMode);

	@Nullable
	PaletteColor getSelectedColor();

	boolean isSelectedColor(@NonNull PaletteColor paletteColor);

	void refreshLastUsedTime();

}

File: OsmAnd/src/net/osmand/plus/card/color/palette/main/v2/IColorsPaletteController.kt
package net.osmand.plus.card.color.palette.main.v2

import android.view.View
import androidx.fragment.app.FragmentActivity
import net.osmand.plus.base.dialog.interfaces.controller.IDialogController
import net.osmand.plus.palette.view.renderer.PaletteItemBinder
import net.osmand.plus.track.fragments.controller.ColorPickerDialogController
import net.osmand.shared.palette.data.PaletteSortMode
import net.osmand.shared.palette.domain.PaletteItem

interface IColorsPaletteController : IDialogController,
	ColorPickerDialogController.ColorPickerListener {

	companion object {
		const val ALL_COLORS_PROCESS_ID = "show_all_colors_screen"
	}

	fun bindPalette(palette: IColorsPalette)
	fun unbindPalette(palette: IColorsPalette)
	fun onAllColorsScreenClosed() {}

	fun setPaletteListener(listener: OnColorsPaletteListener?)

	fun getControlsAccentColor(nightMode: Boolean): Int
	fun isAccentColorCanBeChanged(): Boolean

	fun onSelectItemFromPalette(item: PaletteItem, renewLastUsedTime: Boolean)

	fun selectPaletteItem(item: PaletteItem?)
	fun selectPaletteItem(color: Int, addIfNoFound: Boolean)

	// --- Adapter interaction ---

	fun onPaletteItemLongClick(activity: FragmentActivity, view: View, item: PaletteItem, nightMode: Boolean)

	fun onAllColorsButtonClicked(activity: FragmentActivity)
	fun onAddColorButtonClicked(activity: FragmentActivity)

	fun getPaletteItems(sortMode: PaletteSortMode): List<PaletteItem>
	fun getSelectedPaletteItem(): PaletteItem?
	fun isPaletteItemSelected(item: PaletteItem): Boolean

	fun refreshLastUsedTime()

	fun getItemBinder(
		activity: FragmentActivity,
		nightMode: Boolean
	): PaletteItemBinder
}